pps_mw_training/pipelines/cloud_base/training_data.py

```diff
@@ -0,0 +1,143 @@
+from pathlib import Path
+import json
+import numpy as np  # type: ignore
+import tensorflow as tf  # type: ignore
+import xarray as xr  # type: ignore
+from pps_mw_training.utils.scaler import get_scaler
+
+
+def _load_data(
+    data_files: np.ndarray,
+    input_parameters: str,
+    label_parameters: str,
+    fill_value_input: float,
+    fill_value_label: float,
+) -> list[np.ndarray]:
+    """Load, scale, and filter data."""
+    with xr.open_mfdataset(
+        [f.decode("utf-8") for f in data_files],
+        combine="nested",
+        concat_dim="nscene",
+    ) as all_data:
+        input_params = json.loads(input_parameters)
+        label_params = json.loads(label_parameters)
+
+        input_data = scale_data(all_data, input_params, fill_value_input)
+        label_data = scale_data(all_data, label_params, fill_value_label)
+        return [
+            np.float32(input_data),
+            np.float32(label_data),
+        ]
+
+
+def scale_data(
+    data: xr.Dataset, params: list[dict[str, str | float]], fill_value: float
+) -> np.ndarray:
+    """Scale data"""
+    scaler = get_scaler(params)
+    data = np.stack(
+        [
+            scaler.apply(data[p["name"]].values, idx)
+            for idx, p in enumerate(params)
+        ],
+        axis=3,
```

Comment:
you could add dtype=np.float32 here and remove the casting in the function above.

Reply:
Fixed